VSRAD.DebugServer/IPC/Responses.cs

```diff
@@ -177,6 +195,134 @@ public void Serialize(IPCWriter writer)
+            writer.Write((byte)Status);
+        }
+    }
+
+    public sealed class PutDirectoryResponse : IResponse
+    {
+        public PutDirectoryStatus Status { get; set; }
+
+        public override string ToString() => string.Join(Environment.NewLine, new[]
+        {
+            "PutDirectoryResponse",
+            $"Status = {Status}",
+        });
+
+        public static PutDirectoryResponse Deserialize(IPCReader reader) => new PutDirectoryResponse
+        {
+            Status = (PutDirectoryStatus)reader.ReadByte()
+        };
+
+        public void Serialize(IPCWriter writer)
+        {
+            writer.Write((byte)Status);
+        }
+    }
+    public sealed class GetFileResponse : IResponse
+    {
+        public GetFileStatus Status { get; set; }
+
+        public override string ToString() => string.Join(Environment.NewLine, new[]
+        {
+            "SendFileResponse",
+            $"Status = {Status}",
+        });
+
+        public static GetFileResponse Deserialize(IPCReader reader) => new GetFileResponse
+        {
+            Status = (GetFileStatus)reader.ReadByte()
+        };
+
+        public void Serialize(IPCWriter writer)
+        {
+            writer.Write((byte)Status);
+        }
+    }
+
+    public sealed class ListFilesResponse : IResponse
+    {
+        public List<FileMetadata> Files { get; set; }
+
+        public override string ToString()
+        {
+            var sb = new StringBuilder();
+            foreach (var file in Files)
+            {
+                sb.AppendLine(file.relativePath_);
+            }
+
+            return string.Join(Environment.NewLine, new[] {
+                "ListFileResponse",
+                $"Files = <{sb.ToString()} >"
+            });
+        }
+        private static XmlSerializer getFormatter()
+        {
+            return new XmlSerializer(typeof(List<FileMetadata>));
+        }
+
+        public static ListFilesResponse Deserialize(IPCReader reader)
+        {
+            List<FileMetadata> files = getFormatter().Deserialize(reader.BaseStream) as List<FileMetadata>;
+            return new ListFilesResponse { Files = files };
+        }
+        public void Serialize(IPCWriter writer)
+        {
+            getFormatter().Serialize(writer.BaseStream, Files);
+        }
+    }
+    public sealed class CheckOutdatedFilesResponse : IResponse
+    {
+        public List<FileMetadata> Files { get; set; }
+
+        public override string ToString()
+        {
+            var sb = new StringBuilder();
+            foreach (var file in Files)
+            {
+                sb.AppendLine(file.relativePath_);
+            }
+
+            return string.Join(Environment.NewLine, new[] {
+                "CheckOutdatedFilesResponse",
+                $"Files = <{sb.ToString()} >"
+            });
+        }
+
+        private static XmlSerializer getFormatter()
+        {
+            return new XmlSerializer(typeof(List<FileMetadata>));
+        }
```

Comment:
Do we really don't have other options to serialize a List of `FileMetadata` but using `XmlSerializer`?

It seems off to me, since we never used such format before anywhere in project.

Also, this declaration clearly suggests that it was originated in some Java code, in C# we use PascalCase in such places.